library/src/c_api/mod.rs

```diff
@@ -46,6 +46,28 @@ pub struct AppParameters {
     pub code_cache_dir: *const libc::c_char,
 }
 
+/// An unknown error occurred while updating. The update was not installed.
+/// This is a catch-all for errors that don't fit into the other categories.
+pub const SHOREBIRD_UPDATE_ERROR: i32 = -1;
+
+/// No update is available (e.g. the app is already up-to-date)
+pub const SHOREBIRD_NO_UPDATE: i32 = 0;
+
+/// An update was installed successfully. It will boot from the update on the
+/// next app launch.
+pub const SHOREBIRD_UPDATE_INSTALLED: i32 = 1;
+
+/// An error occurred while updating. The update was not installed.
+pub const SHOREBIRD_UPDATE_HAD_ERROR: i32 = 2;
+
+/// The downloaded patch was not installed because it was invalid.
+pub const SHOREBIRD_UPDATE_IS_BAD_PATCH: i32 = 3;
```

Comment:
```suggestion
/// A new patch is available for download but was not downloaded because it previously failed to boot.
pub const SHOREBIRD_UPDATE_IS_BAD_PATCH: i32 = 3;
```